docker/reference/reference.go

```diff
@@ -102,13 +107,13 @@ func WithTag(name Named, tag string) (NamedTagged, error) {
 
 // WithDigest combines the name from "name" and the digest from "digest" to form
 // a reference incorporating both the name and the digest.
-func WithDigest(name Named, digest digest.Digest) (Canonical, error) {
-	r, err := distreference.WithDigest(name, digest)
-	if err != nil {
-		return nil, err
-	}
-	return &canonicalRef{namedRef{r}}, nil
-}
+//func WithDigest(name Named, digest digest.Digest) (Canonical, error) {
```

Comment:
This is because of

```
docker/reference/reference.go:106: cannot use digest (type "github.com/opencontainers/go-digest".Digest) as type "github.com/docker/distribution/vendor/github.com/opencontainers/go-digest".Digest in argument to reference.WithDigest
```, right?

I am very confused how this happens. `docker/distribution/reference` imports as `"github.com/opencontainers/go-digest"`, so why does Golang decide to rewrite using the `/vendor` path (_and_ then prohibit naming that path, per https://github.com/golang/go/issues/12111 )?

As a matter of principle, I’d generally prefer not to deviate `reference.go` from the upstream original if possible, though I can’t see a way right now either.

/me delves into Go reference documentation…